src/libical/icalmemory.c

```diff
@@ -281,19 +284,25 @@ char *icalmemory_strdup(const char *s)
     return res;
 }
 
-#if defined(ICALMEMORY_DEFAULT_MALLOC) && !defined(S_SPLINT_S)
+#if defined(MEMORY_CONSISTENCY)
+static icalmemory_malloc_f global_icalmem_malloc = &test_malloc;
```

Comment:
Would it make sense to make the distinction between test versions and regular versions in config.h.cmake? Then we wouldn't have to deal with MEMORY_CONSISTENCY details here. But then we'd have to think about how to `#include "test-malloc.h"`. Not sure which way'd be better.

Reply:
yes, I started moving test-malloc directly into libical  but the patch because a mess.